### PR TITLE
[feat] torch-trt: set_multi_device_safe_mode& support dynamic shape in sequence

### PR DESCRIPTION
**Pull request by yjjinjie**

### CODE REVIEW

File: tzrec/tests/train_eval_export_test.py [hunk 1]
```diff
@@ -748,9 +755,9 @@ def test_multi_tower_with_fg_train_eval_export_trt(self):
             )
             df = df.sort_values(by=list(df.columns)).reset_index(drop=True)
             df_t = df_t.sort_values(by=list(df_t.columns)).reset_index(drop=True)
-            # self.assertTrue(df.equals(df_t))
             print(df)
```
Comment (tiankongdeguiji) [inline tzrec/tests/train_eval_export_test.py]: remove the print

File: tzrec/tests/train_eval_export_test.py [hunk 2]
```diff
@@ -781,9 +788,9 @@ def test_multi_tower_with_fg_train_eval_export_trt(self):
...
```
Comment (tiankongdeguiji) [inline tzrec/tests/train_eval_export_test.py]: remove the print

File: tzrec/acc/utils.py [hunk 3]
```diff
@@ -175,3 +177,13 @@ def lists_are_equal(list1: List[torch.Tensor], list2: List[torch.Tensor]) -> boo
         if not torch.equal(list1[i], list2[i]):
             return False
     return True
+
+
+def is_close(df1: pd.DataFrame, df2: pd.DataFrame, abs_tol: float) -> bool:
```
Comment (tiankongdeguiji) [inline tzrec/acc/utils.py]: move to tzrec/utils/test_util.py and rename to df_is_close is better

File: tzrec/main.py [hunk 4]
```diff
@@ -1061,16 +1087,16 @@ def _write(
...
-            v = predictions[c]
-            v = v.tolist() if v.ndim > 1 else v.numpy()
-            output_dict[c] = pa.array(v)
         reserve_batch_record = reserves.get()
         if reserve_batch_record is not None:
             for k, v in zip(
                 reserve_batch_record.column_names, reserve_batch_record.columns
             ):
                 output_dict[k] = v
+        for c in output_cols:
```
Comment (tiankongdeguiji) [inline tzrec/main.py]: why move to here?

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: tzrec/tests/train_eval_export_test.py [hunk 5]
```diff
@@ -781,9 +785,8 @@ def test_multi_tower_with_fg_train_eval_export_trt(self):
             )
             df = df.sort_values(by=list(df.columns)).reset_index(drop=True)
             df_t = df_t.sort_values(by=list(df_t.columns)).reset_index(drop=True)
-            # self.assertTrue(df.equals(df_t))
-            print(df)
-            print(df_t)
+            # differences = df.compare(df_t)
+            # self.assertTrue(dfs_are_close(df, df_t, 1e-6))
```
Comment (tiankongdeguiji) [inline tzrec/tests/train_eval_export_test.py:790]: Are predictions not close now?

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: tzrec/tests/train_eval_export_test.py (hunk 5 above)
Comment (yjjinjie) [reply tzrec/tests/train_eval_export_test.py:790]: yes. continue to solve

File: tzrec/main.py (hunk 4 above)
Comment (yjjinjie) [reply tzrec/main.py]: the accuracy is so close, the sorting of  user_id,item_id may  be incorrect.

**Review by tiankongdeguiji — APPROVED** [review]
LGTM

**Pull request merged**